server/pbench/bin/index-pbench

```diff
@@ -287,20 +315,39 @@ class ResultData(object):
                 self.counters['not_valid_json_file'] += 1
                 continue
 
-            convert_to_float(results)
+            # There should always be a 'parameters' block.
+            # There is no reason for this to be an array, but it is: we
+            # just pick the 0^th (and only element) as the metadata for
+            # the document.
+            assert len(results['parameters']['benchmark'])
+            md = results['parameters']['benchmark'][0]
+            md.update({"iteration_name": iter_name,
+                       "iteration_number": iter_number,
+                       "sample": sample})
+            benchmark = md['benchmark_name']
+            # for each file, generate a sequence of documents and yield each one
+            for ts, src in ResultData.gen_sample_sources(results, benchmark):
+                # each generated result is a document, accompanied by a timestamp
+                # and some metadata.
+                source = _dict_const()
+                source['@metadata'] = self.run_metadata
+                source['@metadata'].update(md)
+
+                # XXX ASSUMPTION: the timestamps we get from the result data are
+                # "milliseconds since epoch" and they are are *assumed* to be UTC.
+                source['@timestamp'] = ts
+                source['results'] = src
 
-            source = _dict_const()
-            source['results'] = results
-            source['@metadata'] = self.run_metadata
-            ts_str = self.run_metadata['runtstamp']
-            source['@timestamp'] = ts_str
+                yield source, source_id
 
-            yield source, source_id
         return
 
     def make_source(self):
-        """Simple jump method to pick the write source generator based on the
-        handler's prospectus."""
+        """
+        If we ever decide we need to process anything other than JSON files, this
+        is going to become a jump method, similar to ToolData.make_source().
+        ATM, we only handle JSON files.
+        """
         if not self.json_files:
             # If we do not have any json files for this experiment, ignore it.
```

Comment:
s/experiment/result/?

Reply:
We use "experiment" in a number of places and I tend to prefer it here:  I think of it as "we run an experiment and we get results, but this experiment produced no results (at least no result.json files)." If you prefer it, I can change it, but please take a look at the other "experiment" instances and see if we need to change it in more places than just here.